### PR TITLE
For #9670: fix intermittent tests and verify system media notification is gone  

### PR DESCRIPTION
**Pull request by sv-ohorvath**
Extends the Media Notification Tests with an extra verification step.
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by AaronMT** [comment]
Did you also want to re-enable the test in this PR and it will re-run again?

**Comment by sv-ohorvath** [comment]
> Did you also want to re-enable the test in this PR and it will re-run again?
@AaronMT 
I can try that. I've modified the waitNotNull condition and re-enabled the test to see how it behaves, so I'll run the checks a few times here before merging. If you can have a look again after the push is complete. Thanks.

**Comment by AaronMT** [comment]
@sv-ohorvath There is another intermittent I am seeing on a WIP branch in `closePrivateTabsNotificationTest` 

https://console.firebase.google.com/u/0/project/moz-fenix/testlab/histories/bh.66b7091e15d53d45/matrices/5335112653220673877/executions/bs.d8ca40b5f5db167d/testcases/3/test-cases

Video here https://storage.cloud.google.com/fenix_test_artifacts/2020-04-07_16-52-45.587000_lMlv/Pixel2-28-en-portrait-shard_48/video.mp4

Maybe it needs a similar style fix, are you able to take a look at it in this PR too?

**Comment by sv-ohorvath** [comment]
> @sv-ohorvath There is another intermittent I am seeing on a WIP branch in `closePrivateTabsNotificationTest`
> 
> https://console.firebase.google.com/u/0/project/moz-fenix/testlab/histories/bh.66b7091e15d53d45/matrices/5335112653220673877/executions/bs.d8ca40b5f5db167d/testcases/3/test-cases
> 
> Video here https://storage.cloud.google.com/fenix_test_artifacts/2020-04-07_16-52-45.587000_lMlv/Pixel2-28-en-portrait-shard_48/video.mp4
> 
> Maybe it needs a similar style fix, are you able to take a look at it in this PR too?

That one looks different because it can't bring focus back to the app after dismissing the notification. I've added a step to tap on the homescreen to force it back into focus. Let's see how it behaves after running the checks for a while.

File: app/src/androidTest/java/org/mozilla/fenix/helpers/TestHelper.kt [hunk 1]
```diff
@@ -89,4 +94,20 @@ object TestHelper {
...
+                view.getLocationOnScreen(screenPos)
+                val screenX = screenPos[0] + x.toFloat()
+                val screenY = screenPos[1] + y.toFloat()
+                floatArrayOf(screenX, screenY)
+            },
+            Press.FINGER,
+            0,
+            0
+        )
+    }
```
[the remark this answers is not in the document]
Comment (sv-ohorvath) [reply app/src/androidTest/java/org/mozilla/fenix/helpers/TestHelper.kt:112]: That method just taps the screen at the given coordinates. Here it is set to a spot where there is nothing on the page. Happy to use something else, but the error with which the test failed seems to be something specific to emulators, I've found discussions about it on Bitrise too, with no solution. I'm happy to remove this change from this PR and keep investigating.

[the remark this answers is not in the document]
Comment (sv-ohorvath) [reply app/src/androidTest/java/org/mozilla/fenix/helpers/TestHelper.kt]: @AaronMT @rpappalax I moved it to a new branch, to be able to move along with this PR
https://github.com/mozilla-mobile/fenix/blob/fix9783-closePrivateTabsNotificationTest/app/src/androidTest/java/org/mozilla/fenix/helpers/TestHelper.kt#L98

**Comment by sv-ohorvath** [comment]
These changes were already approved by Aaron and Richard. Tests have passed on every run (~12 times). I'll go ahead and merge it.

**Pull request merged**